### PR TITLE
Update deps to bring in support for OpenBSD

### PR DESCRIPTION
**Pull request by qbit**
#### Name of feature:

Let `step-ca` build on OpenBSD.

With these deps bumped, tests all pass on OpenBSD amd64.

#### Supporting links/other PRs/issues:

OpenBSD support was added to the following deps:

 - github.com/go-piv/piv-go in https://github.com/go-piv/piv-go/pull/101
 - github.com/newrelic/go-agent in https://github.com/newrelic/go-agent/pull/455
 - github.com/miekg/pkcs11 in https://github.com/miekg/pkcs11/pull/140

### CODE REVIEW

**Comment by tashian** [comment]
Thanks for putting this together @qbit!

File: go.mod
Comment (maraino) [inline go.mod:38]: It looks like the commit https://github.com/newrelic/go-agent/commit/aee135dc5dcf93df9dec7d55ab6ae6d40c0a4d7b is tagged with `v3.18.0`, can you use that instead?

File: go.mod
Comment (maraino) [inline go.mod:23]: I've noticed you created the PR for supporting OpenBSD on `piv-go`. I will accept this version, but can you request `piv-go` maintainers to create a new release?

**Review by maraino — CHANGES_REQUESTED** [review]
@qbit Thanks for this, but can you change the version of the go-agent and if possible get a new release of go-piv.

File: go.mod
Comment (qbit) [reply go.mod]: https://github.com/go-piv/piv-go/issues/105 Done.

File: go.mod
Comment (qbit) [reply go.mod:23]: Done: https://github.com/go-piv/piv-go/issues/105

**Comment by qbit** [comment]
OK, updated go-agent and made a PR to go-piv for cutting a new version.

**Comment by qbit** [comment]
yay! go-piv has a new release, I'll update to include that.

**Comment by qbit** [comment]
Ok, both updates are pointing to tagged releases!

**Review by maraino — APPROVED** [review]
Thanks @qbit.

I'll update https://github.com/smallstep/step-kms-plugin with that version of piv-go and pkcs11. Check it out, is a plugin to `step` to use `step certificate create` and `step certificate sign` with PKCS#11 modules, yubikeys, CloudKMS, ...

**Comment by maraino** [comment]
@qbit I'm gonna take a look at what's going on with the linker before merging ...

**Pull request merged**
